lib/metadata.js

```diff
@@ -737,24 +738,46 @@ TableMetadata.prototype.build = function (tableRow, columnRows) {
     };
     this.columns.push(c);
     columnsKeyed[c.name] = c;
+    switch (row['type']) {
```

Comment:
There seems to be a slight regression with 'isCompact' here.  'tbl6' in metadata-tests.js is not being listed as compact when it should (Metadata #getTable() should retrieve the metadata of a compact storaged table).  Looking into it.